cmd/kubectl-gadget/utils.go

```diff
@@ -131,3 +155,225 @@ func randomTraceID() string {
+		LabelSelector: fmt.Sprintf("trace-template-hash=%s", traceID),
+		FieldSelector: fields.Everything().String(),
+	}
+
+	// Wait until results are ready and fetch all results
+	// TODO: use a watcher to avoid looping client-side
+	//
+	// watch, err := traceRestClient.
+	//	Get().
+	//	Namespace("gadget").
+	//	Resource("traces").
+	//	VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+	//	Watch(context.TODO())
+	// if err != nil {
+	//	contextLogger.Fatalf("Error waiting for traces: %q", err)
+	// }
+	// for event := range watch.ResultChan() {
+	//	fmt.Printf("Event: %v\n", event)
+	//	if data, ok := event.Object.(*metav1.Status); ok {
+	//		contextLogger.Infof("watcher status: %s", data.Message)
+	//	} else if t, ok := event.Object.(*gadgetv1alpha1.Trace); ok {
+	//		fmt.Printf("Got: %v\n", t)
+	//	} else {
+	//		contextLogger.Fatalf("Error waiting for traces: got unexpected %v", event)
+	//	}
+	// }
+
+	var results gadgetv1alpha1.TraceList
+	start := time.Now()
+RetryLoop:
+	for {
+		results = gadgetv1alpha1.TraceList{}
+		err = traceRestClient.
+			Get().
+			Namespace("gadget").
+			Resource("traces").
+			VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+			Do(context.TODO()).
+			Into(&results)
+		if err != nil {
+			deleteTraces(contextLogger, traceRestClient, traceID)
+			contextLogger.Fatalf("Error getting traces: %q", err)
+		}
+
+		timeout := time.Now().Sub(start) > 2*time.Second
+		successNodeCount := 0
+		nodeErrors := make(map[string]string)
+		for _, i := range results.Items {
+			if i.Status.State == "Completed" || i.Status.State == "Started" {
+				successNodeCount++
+			} else {
+				if timeout {
+					nodeErrors[i.Spec.Node] = i.Status.OperationError
+					continue
+				}
+				time.Sleep(100 * time.Millisecond)
+				continue RetryLoop
+			}
+		}
+		for node, err := range nodeErrors {
+			contextLogger.Warningf("Error getting traces from node %q: %s", node, err)
+		}
+		if successNodeCount == 0 {
+			deleteTraces(contextLogger, traceRestClient, traceID)
+			contextLogger.Fatalf("Error getting traces from all nodes")
+		}
+		break RetryLoop
+	}
+
+	if callback == nil {
+		genericStreamsDisplay(contextLogger, client, params, &results)
+	} else {
+		callback(contextLogger, nodes, &results)
+	}
+	deleteTraces(contextLogger, traceRestClient, traceID)
+}
+
+func genericStreamsDisplay(contextLogger *log.Entry, client *kubernetes.Clientset, params *utils.CommonFlags, results *gadgetv1alpha1.TraceList) {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	failure := make(chan string)
+
+	for _, i := range results.Items {
+		if params.Node != "" && i.Spec.Node != params.Node {
+			continue
+		}
+		go func(nodeName, namespace, name string) {
+			cmd := fmt.Sprintf("exec gadgettracermanager -call receive-stream -tracerid trace_%s_%s",
+				namespace, name)
+			var err error
+			err = execPod(client, nodeName, cmd, os.Stdout, os.Stderr)
```

Comment:
Is it ok to pass stout directly? shouldn't it use `postProcess` (or something similar) to avoid "interleaving" lines coming from different nodes?